lib/sigstore/signer.rb

```diff
@@ -169,82 +227,51 @@ def submit_signed_metadata_to_transparency_service(signature, cert, data_sha256)
       # The signing metadata might contain additional, application-specific metadata according to the format used.
       # The Signer then canonically encodes the metadata (according to the chosen format).
 
+      # TODO: allow configuring the entry kind?
+      proposed_entry = build_proposed_hashed_rekord_entry(signature, cert, data_sha256)
+
       # TODO: is looping here correct?
       @trusted_root.tlogs_for_signing.map do |ctlog|
         logger.info { "Submitting to #{ctlog.base_url}" }
 
-        body = {
-          "spec" => {
-            "signature" => {
-              "content" => Internal::Util.base64_encode(signature),
-              "publicKey" => {
-                "content" => Internal::Util.base64_encode(cert.to_pem)
-              }
-            },
-            "data" => {
-              "hash" => {
-                "algorithm" => "sha256", # TODO: should this always be sha256?
-                "value" => Internal::Util.hex_encode(data_sha256)
-              }
-            }
-          },
-          "kind" => "hashedrekord", # TODO: is hashedrekord always the right kind? should this be configurable?
-          "apiVersion" => "0.0.1"
-        }
-
-        resp = Net::HTTP.post(URI.join(ctlog.base_url, "api/v1/log/entries"), body.to_json, {
-                                "Content-Type" => "application/json"
-                              })
-
-        logger.debug do
-          "#{resp.code} #{resp.message.inspect}\n\n#{JSON.pretty_generate(body)}\n\n#{resp.body}"
-        end
-
-        unless resp.code == "201"
-          raise Error::Signing, "#{resp.code} #{resp.message.inspect}\n\n#{JSON.pretty_generate(body)}\n\n#{resp.body}"
-        end
-
-        body = JSON.parse(resp.body)
-
         # TODO: verify
         # The signer MUST verify the log entry as in Spec: Transparency Service.
```

Comment:
given this happens _immediately_ below in verify, is verifying the log entry _separately_ necessary?

Reply:
(maybe i'm reading this ruby code wrong but) Java does a verification here https://github.com/sigstore/sigstore-java/blob/main/sigstore-java/src/main/java/dev/sigstore/KeylessSigner.java#L338 immediately after obtaining the log entry. There's some presumed minor benefit that if the log goes rogue, you get to detect it a little early?

Reply:
sigstore-go does not verify the transparency log entry it receives during signing, but it will verify the bundle before returning it to the user, if a trusted root to verify with is provided.